composeApp/src/commonMain/kotlin/login/LoginScreen.kt

```diff
@@ -0,0 +1,104 @@
+package login
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import designsystem.colors.ButtonColor
+import designsystem.login.CredentialInput
+import designsystem.login.FacebookSignUpHyperlink
+import designsystem.login.ForgotPasswordHyperlink
+import designsystem.login.InstagramTextLogo
+import designsystem.login.InstagramSignUpHyperlink
+import designsystem.login.LoginButton
+import designsystem.login.LoginDivider
+import designsystem.login.LoginFooter
+import dev.icerock.moko.resources.compose.painterResource
+import dev.icerock.moko.resources.compose.stringResource
+import instaclone.resources.MR
+
+val horizontalPadding = 16.dp
+
+@Composable
+fun LoginScreen() {
```

Comment:
I think we should create a LoginScreen using Voyager here. Something like:

```kotlin
class LoginScreen : Screen {
    @Composable
    override fun Content() {
        val viewModel = rememberScreenModel { LoginViewModel() }
        val uiState by viewModel.uiState.collectAsState()

        return LoginScreenContent(
            uiState = uiState,
            onLoginEvent = viewModel::onLoginEvent
        )
    }
}
```

- Your `LoginScreen` function will be `LoginScreenContent`;

- Your `ViewModel` will be something like:

```kotlin
class LoginViewModel(
    private val repository: LoginRepository = LoginRepositoryImpl()
) : ScreenModel {
    private val _state = MutableStateFlow(LoginUIState())
    val uiState: StateFlow<LoginUIState>
        get() = _state

   fun onLoginEvent() {...}
}
```

Reply:
We can do a pair programming if you want 🤗

Reply:
Yep, that's the one that got me hahahaha. I will take you up on that since I don't know the concepts you presented above